src/SAML2/WebSSO/Types.hs

```diff
@@ -1,7 +1,140 @@
 {-# LANGUAGE StrictData          #-}
 {-# LANGUAGE OverloadedStrings   #-}
 
-module SAML2.WebSSO.Types where
+module SAML2.WebSSO.Types
+  ( XmlText, mkXmlText, escapeXmlText
+  , AccessVerdict(..)
+  , _AccessDenied
+  , _AccessGranted
+  , avReasons
+  , avUserId
+  , DeniedReason(..)
+  , UserRef(..)
+  , uidTenant, uidSubject
+  , Issuer(..)
+  , fromIssuer
+  , SPMetadata(..)
+  , spID
+  , spValidUntil
+  , spCacheDuration
+  , spOrgName
+  , spOrgDisplayName
+  , spOrgURL
+  , spResponseURL
+  , spContacts
+  , ContactPerson(..)
+  , cntType
+  , cntCompany
+  , cntGivenName
+  , cntSurname
+  , cntEmail
+  , cntPhone
+  , ContactType(..)
+  , IdPMetadata(..)
+  , edIssuer
+  , edRequestURI
+  , edCertAuthnResponse
+  , IdPId(..)
+  , IdPConfig_
+  , IdPConfig(..)
+  , idpId
+  , idpMetadata
+  , idpExtraInfo
+  , AuthnRequest(..)
+  , rqID
+  , rqIssueInstant
+  , rqIssuer
+  , rqNameIDPolicy
+  , Comparison(..)
+  , RequestedAuthnContext(..)
+  , rqacAuthnContexts
+  , rqacComparison
+  , NameIdPolicy(..)
+  , nidFormat
+  , nidSpNameQualifier
+  , nidAllowCreate
+  , AuthnResponse
+  , Response(..)
+  , rspID
+  , rspInRespTo
+  , rspIssueInstant
+  , rspDestination
+  , rspIssuer
+  , rspStatus
+  , rspPayload
+  , Time(..), timeFormat, addTime
+  , Duration(..)
+  , ID(..), mkID
+  , BaseID, mkBaseID
+  , baseID
+  , baseIDNameQ
+  , baseIDSPNameQ
+  , NameID, mkNameID, unspecifiedNameID, emailNameID, entityNameID
+  , nameID
+  , nameIDNameQ
+  , nameIDSPNameQ
+  , nameIDSPProvidedID
+  , nameIDToST
+  , shortShowNameID
+  , NameIDFormat(..), nameIDFormat
+  , NameIDReprFormat
+  , Email(..)
+  , UnqualifiedNameID(..)
+  , mkUNameIDUnspecified
+  , mkUNameIDEmail
+  , mkUNameIDX509
+  , mkUNameIDWindows
+  , mkUNameIDKerberos
+  , mkUNameIDEntity
+  , mkUNameIDPersistent
+  , mkUNameIDTransient
+  , unameIDFormat
+  , Status(..)
+  , Assertion(..)
+  , assID
+  , assIssueInstant
+  , assIssuer
+  , assConditions
+  , assContents
+  , Conditions(..)
+  , condNotBefore
+  , condNotOnOrAfter
+  , condOneTimeUse
+  , condAudienceRestriction
+  , SubjectAndStatements(..)
+  , sasSubject
+  , sasStatements
+  , Subject(..)
+  , subjectID
+  , subjectConfirmations
+  , SubjectConfirmation(..)
+  , scMethod
+  , scData
+  , SubjectConfirmationMethod(..)
+  , SubjectConfirmationData(..)
+  , scdNotBefore
+  , scdNotOnOrAfter
+  , scdRecipient
+  , scdInResponseTo
+  , scdAddress
+  , IP, ipToST, mkIP
+  , DNSName, fromDNSName, mkDNSName
+  , Statement(..), isAuthnStatement
+  , astAuthnInstant
+  , astSessionIndex
+  , astSessionNotOnOrAfter
+  , astSubjectLocality
+  , Locality(..)
+  , localityAddress
+  , localityDNSName
+  , normalizeAssertion
+  , idPIdToST
+  , assEndOfLife
+  , rspInResponseTo
+  , getUserRef
+  , nelConcat
+  , (<$$>)
```

Comment:
I have considered moving `XmlText` to a new module so I can skip the explicit exports in this module, but decided that explicit exports are nice, even if they are a bit long.  (Nothing I feel too strongly about, though.)

Reply:
I like them and would even enforce them on the project-wide level. Having sections would also be nice, though.